### PR TITLE
♻️ ✨ Cleanup director refs and get image manifest digest (part 3)

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

This PR does some cleanup before extending the service syncer.

- 🗑️ Removes old and unused references to the director
- ✨ `director` now returns [image manifest digest](https://distribution.github.io/distribution/spec/api/#content-digests) 
      - `GET /v0/services/simcore%2Fservices%2Fcomp%2Fans-model/1.0.0/labels` responds with image manifest digest value at `io.simcore.image.digest`
```json
{
"data": {
    "io.simcore.key": "{\"key\": \"simcore/services/comp/ans-model\"}",
    # ...
    "image_digest": "sha256:bebc4af23d55a3a40c7c291181eff5ce6974dec824d0d739422f38ebb0bda674"
  }
}
```
- 🗑️ Removes old and unused references to the `dag` table
- ♻️ Unified errors in `catalog`

## Related issue/s

- Follow up from https://github.com/ITISFoundation/osparc-simcore/pull/6027
- Implements part of #5964

## How to test

- Driving test `services/director/tests/test_registry_proxy.py`

## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

The `DIRECTOR_*` env-var ONLY used in the `catalog` service and in the `director-v2`

### CODE REVIEW

File: services/docker-compose.yml [hunk 1]
```diff
@@ -590,9 +590,6 @@ services:
       DIRECTOR_V2_HOST: ${DIRECTOR_V2_HOST}
       DIRECTOR_V2_PORT: ${DIRECTOR_V2_PORT}
 
-      # WEBSERVER_DIRECTOR
-      DIRECTOR_HOST: ${DIRECTOR_HOST}
```
Comment (YuryHrytsuk) [inline services/docker-compose.yml:594]: Are these 2 ENV now obsolete and need to be removed from `repo.config`?

**Review by YuryHrytsuk — APPROVED** [review]
Thanks!

**Review by odeimaiz — APPROVED** [review]
Full Stack!

File: services/docker-compose.yml (hunk 1 above)
Comment (pcrespov) [reply services/docker-compose.yml:594]: @YuryHrytsuk No, they are not obsolete. That is what I tried to explain in the description above. The are used only in two services (director-v2 and catalog) but they were exposed to five!

**Review by sanderegg — APPROVED** [review]

**Review by matusdrobuliak66 — APPROVED** [review]
:100:

**Review by GitHK — APPROVED** [review]

File: packages/models-library/src/models_library/services_metadata_published.py [hunk 2]
```diff
@@ -161,6 +161,12 @@ class ServiceMetaDataPublished(ServiceKeyVersion, ServiceBase):
         description="regexp pattern for detecting computational service's progress",
     )
 
+    # SEE https://github.com/opencontainers/image-spec/blob/main/annotations.md#pre-defined-annotation-keys
+    image_digest: str | None = Field(
+        None,
+        description="Image manifest digest. Note that this is NOT injected as an image label",
+    )
+
```
Comment (bisgaard-itis) [inline packages/models-library/src/models_library/services_metadata_published.py:169]: Over a ☕ someday I would like if you could explain to me this. 😄

**Review by bisgaard-itis — APPROVED** [review]
Very cool! Thanks a lot for the effort

**Review by jsaq007 — APPROVED** [review]
👍🏼

**Pull request merged**